src/interactive-window/debugger/jupyter/debugCellControllers.ts

```diff
@@ -26,26 +27,16 @@ export class DebugCellController implements IDebuggingDelegate {
     public async willSendEvent(_msg: DebugProtocolMessage): Promise<boolean> {
         return false;
     }
-
     public async willSendRequest(request: DebugProtocol.Request): Promise<void> {
         const metadata = getInteractiveCellMetadata(this.debugCell);
-        if (request.command === 'configurationDone' && metadata && metadata.generatedCode) {
+        if (request.command === 'setBreakpoints' && metadata && metadata.generatedCode && !this.cellDumpInvoked) {
+            this.cellDumpInvoked = true;
             await cellDebugSetup(this.kernel, this.debugAdapter);
-
-            const realPath = this.debugAdapter.getSourcePath(metadata.interactive.uristring);
```

Comment:
No more adding of breakpoints, we use the technique of injecting `breakpoint()`, that works much better.